tests/api/nodes_test.py

```diff
@@ -61,3 +61,249 @@ def test_read_nodes(session: Session, client: TestClient) -> None:
             "type": "INT",
         },
     ]
+
+
+class TestValidateNodes:  # pylint: disable=too-many-public-methods
```

Comment:
do you think we should work on a universal fixture for these sessions? right now there's the `construction_session` maybe we can merge them

Reply:
Great suggestion. If it's ok I'll merge them into a global fixture in a follow-up PR. Only because it'll be pretty noisy to add to this PR's diff.